apps/cli/src/transforms/images.ts

```diff
@@ -45,50 +52,105 @@ async function downloadAndSave(session: ISession, url: string, file: string): Pr
-      let file: string | undefined;
-      if (oxa) {
-        // If oxa, get the download url
-        const versionId = oxa?.block as VersionId;
-        if (!versionId?.version) return;
-        const url = versionIdToURL(versionId);
-        session.log.debug(`Fetching image version: ${url}`);
-        const { ok, json } = await session.get(url);
-        const downloadUrl = json.links?.download;
-        if (!ok || !downloadUrl) {
-          session.log.error(`Error fetching image version: ${url}`);
-          return;
-        }
-        file = await downloadAndSave(
-          session,
-          downloadUrl,
-          `${versionId.block}.${versionId.version}`,
-        );
-      } else if (isUrl(sourceUrl)) {
-        // If not oxa, download the URL directly and save it to a file with a hashed name
-        file = await downloadAndSave(session, sourceUrl, computeHash(sourceUrl));
-      } else if (fs.existsSync(imageLocalFile)) {
-        // Non-oxa, non-url local image paths relative to the config.section.path
-        file = hashAndCopyStaticFile(session, imageLocalFile);
-        if (!file) return;
-      } else if (isBase64(sourceUrl)) {
-        // Inline base64 images
-        const fileObject = new WebFileObject(session.log, staticPath(session), '', true);
-        await fileObject.writeBase64(sourceUrl);
-        file = fileObject.id;
-      } else {
-        session.log.error(`Cannot find image "${sourceUrl}" in ${filePath}`);
-        return;
+      const result = await saveImageInStaticFolder(
+        session,
+        image.sourceUrl || image.url,
+        dirname(file),
+      );
+      if (result) {
+        // Update mdast with new file name
+        const { sourceUrl, url } = result;
+        image.sourceUrl = sourceUrl;
+        image.url = url;
       }
-      // Update mdast with new file name
-      image.sourceUrl = sourceUrl;
-      image.url = `/_static/${file}`;
     }),
   );
 }
+
+export async function transformThumbnail(
+  session: ISession,
+  frontmatter: PageFrontmatter,
+  mdast: Root,
+  file: string,
+) {
+  let thumbnail = frontmatter.thumbnail;
+  // If the thumbnail is explicitly null, don't add an image
+  if (thumbnail === null) {
+    session.log.debug(`${file}#frontmatter.thumbnail is explicitly null, not searching content.`);
+    return;
+  }
+  if (!thumbnail) {
+    // The thumbnail isn't found, grab it from the mdast
+    const [image] = selectAll('image', mdast) as GenericNode[];
+    if (!image) {
+      session.log.debug(`${file}#frontmatter.thumbnail is not set, and there are no images.`);
+      return;
+    }
+    session.log.debug(`${file}#frontmatter.thumbnail is being populated by the first image.`);
+    thumbnail = image.sourceUrl || image.url;
+  }
+  if (!thumbnail) return;
+  session.log.debug(`${file}#frontmatter.thumbnail Saving thumbnail in static folder.`);
+  const result = await saveImageInStaticFolder(session, thumbnail, dirname(file));
```

Comment:
Ah, so on markdown export, we copy thumbnails to `thumbnails` folder and rewrite frontmatter, then on transform, we coppy to `_static` and rewrite frontmatter again. I think this all makes sense.